### DISCUSSION TITLE
Test suite

### DISCUSSION THREAD

**Discussion by larsgw** [kdl-org/kdl#50] [Ideas]
Does it make sense to make a test suite for the specification that can be used for different implementations? I was planning to make one for kdljs anyway... I guess the serialization of the parsed KDL is a bit difficult to standardize across implementations though, especially things like numbers with varying precision.

**Comment by Andy-Bell** [comment]
I have started toying about with writing a parser in ruby and for now I am just using the current examples as my tests, a proper suite would be amazing

**Comment by larsgw** [comment]
See also https://github.com/kdl-org/kdl/discussions/78
